persistence-modules/spring-data-jpa-filtering/src/test/java/com/baeldung/projection/JpaProjectionIntegrationTest.java

```diff
@@ -57,4 +61,13 @@ public void whenUsingDynamicProjections_thenObjectWithRequiredPropertiesIsReturn
         assertThat(personView.getFirstName()).isEqualTo("John");
         assertThat(personDto.getFirstName()).isEqualTo("John");
     }
+    
+    @Test
+    public void givenDTOFromTwoTables_whenUsingClassBasedProjections_thenDtoWithRequiredPropertiesIsReturned() {
+        List<PersonAddressDTO> dtos = personRepository.findAllPersonsWithAddress();
+        assertEquals(1, dtos.size());
```

Comment:
If you're using `assertJ`'s `assertThat`, then change this to `assertThat(dtos).hasSize(1)`